include/motor-control/firmware/stepper_motor/motor_hardware.hpp

```diff
@@ -53,13 +54,25 @@ class MotorHardware : public StepperMotorHardwareIface {
     void set_LED(bool status) final;
     auto get_encoder_pulses() -> int32_t final;
     void reset_encoder_pulses() final;
-    auto has_cancel_request() -> bool final {
-        return cancel_request.exchange(false);
-    }
     void disable_encoder() final;
     void enable_encoder() final;
 
-    void request_cancel() final { cancel_request.store(true); }
+    // need any std::optional usage?
+    auto has_cancel_request() -> CancelRequest final {
```

Comment:
IMO here this one wants to be called `get_cancel_request`, `has_cancel_request` really makes me think it returns a bool